tests/integration_test/jwt_auth_integration_test.go

```diff
@@ -239,3 +242,93 @@ func TestInvalidOpenIDConnectDiscovery(t *testing.T) {
 		}
 	}
 }
+
+func TestFrontendAndBackendAuthHeaders(t *testing.T) {
+	t.Parallel()
+
+	s := env.NewTestEnv(comp.TestFrontendAndBackendAuthHeaders, platform.EchoRemote)
+	s.OverrideAuthentication(&confpb.Authentication{
+		Rules: []*confpb.AuthenticationRule{
+			{
+				Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.dynamic_routing_BearertokenConstantAddress",
+				Requirements: []*confpb.AuthRequirement{
+					{
+						ProviderId: testdata.TestAuthProvider,
+						Audiences:  "ok_audience",
+					},
+				},
+			},
+			{
+				Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.dynamic_routing_DisableAuthSetToTrue",
+				Requirements: []*confpb.AuthRequirement{
+					{
+						ProviderId: testdata.TestAuthProvider,
+						Audiences:  "ok_audience",
+					},
+				},
+			},
+		},
+	})
+	s.OverrideMockMetadata(
+		map[string]string{
+			util.IdentityTokenSuffix + "?format=standard&audience=https://localhost/bearertoken/constant": "ya29.BackendAuthToken",
+		}, 0)
+
+	defer s.TearDown()
+	if err := s.Setup(utils.CommonArgs()); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	testData := []struct {
+		desc        string
+		method      string
+		path        string
+		headers     map[string]string
+		wantHeaders map[string]string
+	}{
+		{
+			desc: "Frontend auth preserves `Authorization` and creates `X-Endpoint-API-UserInfo`." +
+				"Backend auth is disabled, so no further header modifications.",
+			method: "GET",
+			path:   "/disableauthsettotrue/constant/disableauthsettotrue",
+			headers: map[string]string{
+				"Authorization": "Bearer " + testdata.Es256Token,
+			},
+			wantHeaders: map[string]string{
+				"Authorization":           "Bearer " + testdata.Es256Token,
+				"X-Endpoint-API-UserInfo": testdata.Es256TokenPayloadBase64,
+			},
+		},
+		{
+			desc: "Frontend auth preserves `Authorization` and creates `X-Endpoint-API-UserInfo`." +
+				"Backend auth then modifies `Authorization` and creates `X-Forwarded-Authorization`.",
```

Comment:
cc @fareeda0, this integration test should validate the request header flow you described in #140